frontend/packages/kubevirt-plugin/src/components/modals/scheduling-modals/node-selector-modal/node-selector-modal.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { Modal } from '@patternfly/react-core';
+import { NodeModel } from '@console/internal/models';
+import { getLabel } from '@console/shared';
+import { FirehoseResult } from '@console/internal/components/utils';
+import { ModalFooter } from '../../modal/modal-footer';
+import { VMLikeEntityKind } from '../../../../types/vmLike';
+import { NodeSelector } from '../../../../types';
+import { getLoadedData, isLoaded, getLoadError } from '../../../../utils';
+import { NodeChecker, LabelsList } from '../shared';
+import { NODE_SELECTOR_MODAL_TITLE, SCHEDULING_REQUIRED_LABEL } from '../shared/consts';
+import './node-selector-modal.scss';
+
+export const NSModal = ({
+  nodes,
+  selector,
+  onLabelAdd,
+  onLabelChange,
+  onLabelDelete,
+  onSubmit,
+  onClose,
+  isOpen,
+  inProgress,
+  errorMessage,
+  showPatchError,
+}: NSModalProps) => {
+  const loadError = getLoadError(nodes, NodeModel);
+  const loadedNodes = getLoadedData(nodes, []);
+
+  const [isLoading, setIsLoading] = React.useState<boolean>(true);
+  const [qualifiedNodes, setQualifiedNodes] = React.useState([]);
+
+  // Node search 1 sec after user stopped typing
+  React.useEffect(() => {
+    const debounceNodeSearch = setTimeout(() => {
+      if (isLoaded(nodes)) {
+        const labels = Object.values(selector).filter(({ key }) => !!key);
+        // look only for schedulable nodes
+        labels.push({
+          key: SCHEDULING_REQUIRED_LABEL,
+          value: 'true',
+        });
+
+        const newNodes = loadedNodes.filter((node) =>
```

Comment:
you can getLabels and check if they are not null first and then reuse the labels object for getting the value by key